pkg/kubelet/container/runtime.go

```diff
@@ -79,7 +79,9 @@ type Runtime interface {
 	SyncPod(pod *api.Pod, apiPodStatus api.PodStatus, podStatus *PodStatus, pullSecrets []api.Secret, backOff *flowcontrol.Backoff) PodSyncResult
 	// KillPod kills all the containers of a pod. Pod may be nil, running pod must not be.
 	// TODO(random-liu): Return PodSyncResult in KillPod.
-	KillPod(pod *api.Pod, runningPod Pod) error
+	// gracePeriodOverride if specified allows the caller to override the pod default grace period.
+	// it is usueful when doing SIGKILL for hard eviction scenarios, or max grace period during soft eviction scenarios.
+	KillPod(pod *api.Pod, runningPod Pod, gracePeriodOverride *int64) error
```

Comment:
I went with int for the following reasons:

*  it was consistent with pod termination grace period seconds (int64)
 * https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L1208
* it was consistent with docker client StopContainer timeout in seconds (int)
 * https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/docker.go#L64
* it was consistent with systemd `TimeoutStopSec`
 * https://www.freedesktop.org/software/systemd/man/systemd.service.html

so basically, taking a duration means we have to remember to convert to second precision which seemed more complicated/error prone